vhost/src/vhost_user/backend_req_handler.rs

```diff
@@ -555,6 +561,26 @@ impl<S: VhostUserBackendReqHandler> BackendReqHandler<S> {
                 let res = self.set_backend_req_fd(files);
                 self.send_ack_message(&hdr, res)?;
             }
+            Ok(FrontendReq::GET_SHARED_OBJECT) => {
+                self.check_proto_feature(VhostUserProtocolFeatures::SHARED_OBJECT)?;
+                self.check_request_size(&hdr, size, hdr.get_size() as usize)?;
+                let msg = self.extract_request_body::<VhostUserSharedMsg>(&hdr, size, &buf)?;
+                let res = self.backend.get_shared_object(msg);
+
+                match res {
+                    Ok(file) => {
+                        let hdr = self.new_reply_header::<VhostUserEmpty>(&hdr, 0)?;
+                        self.main_sock.send_message(
+                            &hdr,
+                            &VhostUserEmpty,
+                            Some(&[file.as_raw_fd()]),
+                        )?;
+                    }
+                    Err(_) => {
+                        self.main_sock.send_message(&hdr, &VhostUserEmpty, None)?;
```

Comment:
`hdr` here is still the old one.

Reply:
Thanks, will fix it. I copied what postcopy_advice() did, I think I should change that aswell in another PR.

Reply:
Yeah, I think so, and I agree on a separated PR, thanks in advance for that ;-)